### DISCUSSION TITLE
Add `Result.all`

### DISCUSSION THREAD

**Discussion by chriskrycho** [true-myth/true-myth#206] [Ideas]
I didn't add it when we originally added `Maybe.all` because it wasn't clear what the operation should be for the `Err` case, but it would be straightforward to add now, and worth doing so that you can have a structure-preserving transformation there. We should decide how to treat that scenario: we don't have a type class like `Monoid` available, so we can't do something clever like have a specialization over monoids that concatenates them but otherwise just pick the first one. We *could* special-case that way for something like `Array`, and/or we could simply allow the user to provide an accumulator-style callback.

**Comment by wodeni** [comment]
This would be very helpful! Also, I've been using this library for writing semantic checkers and running into cases like this:
```ts
const type1: Result<T, CheckerError> = checkerFunction1();
const type2: Result<T, CheckerError> = checkerFunction2();
// If both are successful, need to check if type1.value === type2.value
return type1.match({
    Ok: t1 =>
      type2.match({
        Ok: t2 => t1 === t2 ? ok() : err(checkerError("something went wrong")),
        Err: (e) => err(e),
      }),
    Err: (e) => err(e),
  });
```

My case seems to require more than `ap()`, because the "transformation function" returns a `Result` as well. I tried currying and use `andThen`, but TS seems to have problems typechecking it. Will `Result.all` be suitable for this kind of use case? I imagine I can just get a `Result<T[], CheckerError>` back from `Result.all(type1, type2)` and then I can safely use `andThen` to do the equality check.

**Comment by hannesborch** [comment]
I already stumbled over this case several times. Finally, I took myself some time to come up with a solution I initially only built for myself, but which can now also be checked in https://github.com/true-myth/true-myth/pull/1070.

**Comment by chriskrycho** [comment]
Thanks to @hannesborch, we shipped `result.transposeAll` and `result.transposeAny` in v9.1.0!

**Reply by hannesborch** [reply]
Hey @chriskrycho, thanks for taking over. And sorry for not being responsive throughout the last weeks, job and vacation kept me "busy". 🙂

**Reply by chriskrycho** [reply]
You’re welcome, and no need to apologize at all—I was in the same boat myself earlier in the summer! This project runs on “when people have time and energy for it” mode. That’s how we have done it for the past 8 years, and it’s how we’ll do it for however many more years we maintain it. 😁
